drivers/serial/uart_mcux_lpuart.c

```diff
@@ -426,7 +427,9 @@ static const struct mcux_lpuart_config mcux_lpuart_##n##_config = {	\
 	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),		\
 	.clock_subsys = (clock_control_subsys_t)DT_INST_CLOCKS_CELL(n, name),\
 	.baud_rate = DT_INST_PROP(n, current_speed),			\
-	IRQ_FUNC_INIT							\
+	IRQ_FUNC_INIT,							\
```

Comment:
The error is related to the comma after IRQ_FUNC_INIT. I suspect the expansion on the particular build to get to this probably already includes a comma.

From log file:

 /workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:430:15: error: expected expression before ',' token

  430 |  IRQ_FUNC_INIT,       \

      |               ^

../../../../../../../include/sys/util_internal.h:67:26: note: in definition of macro '__DEBRACKET'

   67 | #define __DEBRACKET(...) __VA_ARGS__

      |                          ^~~~~~~~~~~

../../../../../../../include/sys/util_internal.h:59:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'

   59 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)

      |  ^~~~~~~~~~~~~~~~~~~~

../../../../../../../include/sys/util_internal.h:54:2: note: in expansion of macro '__COND_CODE'

   54 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)

      |  ^~~~~~~~~~~

../../../../../../../include/sys/util_macro.h:117:2: note: in expansion of macro 'Z_COND_CODE_1'

  117 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)

      |  ^~~~~~~~~~~~~

../../../../../../../include/devicetree.h:2211:2: note: in expansion of macro 'COND_CODE_1'

 2211 |  COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \

      |  ^~~~~~~~~~~

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:421:2: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG'

  421 |  LPUART_MCUX_DECLARE_CFG(n, LPUART_MCUX_IRQ_CFG_FUNC_INIT)

      |  ^~~~~~~~~~~~~~~~~~~~~~~

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:452:2: note: in expansion of macro 'LPUART_MCUX_INIT_CFG'

  452 |  LPUART_MCUX_INIT_CFG(n);

      |  ^~~~~~~~~~~~~~~~~~~~

zephyr/include/generated/devicetree_unfixed.h:9163:53: note: in expansion of macro 'LPUART_MCUX_INIT'

 9163 | #define DT_FOREACH_OKAY_INST_nxp_kinetis_lpuart(fn) fn(0)

      |                                                     ^~

../../../../../../../include/sys/util_internal.h:85:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_kinetis_lpuart'

   85 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__

      |                                    ^

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:454:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'

  454 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT)

      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~

Reply:
Thanks, I switch the order of flow_ctrl and irq_config_func to put IRQ_FUNC_INIT in the last.